common/util/array.ts

```diff
@@ -1,23 +1,13 @@
-import { isEqual } from 'lodash'
+import { compact, flattenDeep, isEqual } from 'lodash'
 
 export function filterDefined<T>(array: (T | null | undefined)[]) {
   return array.filter((item) => item !== null && item !== undefined) as T[]
 }
 
-export function buildArray<T>(
-  ...params: (T | T[] | false | undefined | null)[]
-) {
-  const array: T[] = []
+type FalseyValueArray<T> = T | false | undefined | null | FalseyValueArray<T>[]
 
-  for (const el of params) {
-    if (Array.isArray(el)) {
-      array.push(...el)
-    } else if (el) {
-      array.push(el)
-    }
-  }
-
-  return array
+export function buildArray<T>(...params: FalseyValueArray<T>[]) {
+  return compact(flattenDeep(params)) as T[]
```

Comment:
Nice. Didn't know about `compact`

Reply:
fyi - can also do this in vanilla js like `params.flat(Infinity).filter(Boolean)` but typescript hates this